server/src/main/java/org/opensearch/action/search/TransportDeletePITAction.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.OpenSearchException;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.StepListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.GroupedActionListener;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.Strings;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.search.SearchService;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.Transport;
+import org.opensearch.transport.TransportService;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+/**
+ * Transport action for deleting pit reader context - supports deleting list and all pit contexts
+ */
+public class TransportDeletePITAction extends HandledTransportAction<DeletePITRequest, DeletePITResponse> {
+    private SearchService searchService;
+    private final NamedWriteableRegistry namedWriteableRegistry;
+    private TransportSearchAction transportSearchAction;
+    private final ClusterService clusterService;
+    private final SearchTransportService searchTransportService;
+    private static final Logger logger = LogManager.getLogger(TransportDeletePITAction.class);
+
+    @Inject
+    public TransportDeletePITAction(
+        SearchService searchService,
+        TransportService transportService,
+        ActionFilters actionFilters,
+        NamedWriteableRegistry namedWriteableRegistry,
+        TransportSearchAction transportSearchAction,
+        ClusterService clusterService,
+        SearchTransportService searchTransportService
+    ) {
+        super(DeletePITAction.NAME, transportService, actionFilters, DeletePITRequest::new);
+        this.searchService = searchService;
+        this.namedWriteableRegistry = namedWriteableRegistry;
+        this.transportSearchAction = transportSearchAction;
+        this.clusterService = clusterService;
+        this.searchTransportService = searchTransportService;
+    }
+
+    @Override
+    protected void doExecute(Task task, DeletePITRequest request, ActionListener<DeletePITResponse> listener) {
+        List<SearchContextIdForNode> contexts = new ArrayList<>();
+        List<String> pitIds = request.getPitIds();
+        if (pitIds.size() == 1 && "_all".equals(pitIds.get(0))) {
```

Comment:
what happens if _all is present along with other pit_ids in a comma separated fashion

Reply:
if passed as part of body, we'll throw 'cannot parse id' found exception